From fe114d110b581336fa03e4da9a296625a7f604c1 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Tue, 22 Sep 2020 12:24:22 -0700
Subject: [PATCH 1647/1852] v4l: xilinx: dprx: Fix warnings for NULL_RETURNS

commit 42aca2edbfb28849691de81bcbab7ee124c98f51 from
https://github.com/Xilinx/linux-xlnx.git

This patch fix the warnings for the NULL_RETURNS type.

The actual warnings are:

For xdprxss_getset_format():
Event returned_null: 	"__xdprxss_get_pad_format" returns "NULL".
Event dereference: 	Dereferencing "__xdprxss_get_pad_format(xdprxss,
cfg, fmt->pad, fmt->which)", which is known to be "NULL"

For __xdprxss_get_pad_format():
Event return_null: 	Explicitly returning null.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-dprxss.c | 23 +++++++++++++++----
 1 file changed, 18 insertions(+), 5 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-dprxss.c b/drivers/media/platform/xilinx/xilinx-dprxss.c
index 6c4127cebde9..57de15580327 100644
--- a/drivers/media/platform/xilinx/xilinx-dprxss.c
+++ b/drivers/media/platform/xilinx/xilinx-dprxss.c
@@ -811,14 +811,21 @@ __xdprxss_get_pad_format(struct xdprxss_state *xdprxss,
 			 struct v4l2_subdev_pad_config *cfg,
 			 unsigned int pad, u32 which)
 {
+	struct v4l2_mbus_framefmt *format;
+
 	switch (which) {
 	case V4L2_SUBDEV_FORMAT_TRY:
-		return v4l2_subdev_get_try_format(&xdprxss->subdev, cfg, pad);
+		format = v4l2_subdev_get_try_format(&xdprxss->subdev, cfg, pad);
+		break;
 	case V4L2_SUBDEV_FORMAT_ACTIVE:
-		return &xdprxss->format;
+		format = &xdprxss->format;
+		break;
 	default:
-		return NULL;
+		format = NULL;
+		break;
 	}
+
+	return format;
 }
 
 /**
@@ -862,6 +869,7 @@ static int xdprxss_getset_format(struct v4l2_subdev *sd,
 				 struct v4l2_subdev_format *fmt)
 {
 	struct xdprxss_state *xdprxss = to_xdprxssstate(sd);
+	struct v4l2_mbus_framefmt *format;
 
 	if (!xdprxss->valid_stream) {
 		dev_err(xdprxss->dev, "Video not locked!\n");
@@ -873,8 +881,13 @@ static int xdprxss_getset_format(struct v4l2_subdev *sd,
 		fmt->format.width, fmt->format.height,
 		fmt->format.code, fmt->format.field,
 		fmt->format.colorspace);
-	fmt->format = *__xdprxss_get_pad_format(xdprxss, cfg,
-						fmt->pad, fmt->which);
+	format = __xdprxss_get_pad_format(xdprxss, cfg,
+					  fmt->pad, fmt->which);
+	if (!format)
+		return -EINVAL;
+
+	fmt->format = *format;
+
 	return 0;
 }
 
-- 
2.31.1

